cmd/del/route53/cmd.go

```diff
@@ -0,0 +1,118 @@
+package route53
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/jharrington22/aws-resource/pkg/arguments"
+	"github.com/jharrington22/aws-resource/pkg/aws"
+	logging "github.com/jharrington22/aws-resource/pkg/logging"
+	rprtr "github.com/jharrington22/aws-resource/pkg/reporter"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+var (
+	zoneId string
+)
+
+var Cmd = &cobra.Command{
+	Use:   "route53",
+	Short: "Delete route53 resources",
+	Long: `Delete route53 resources"
+aws-resource delete route53.`,
+	RunE: run,
+}
+
+func run(cmd *cobra.Command, args []string) (err error) {
+
+	reporter := rprtr.CreateReporterOrExit()
+	logging := logging.CreateLoggerOrExit(reporter)
+
+	awsClient, err := aws.NewClient().
+		Logger(logging).
+		Profile(arguments.Profile).
+		RoleArn(arguments.RoleArn).
+		Region(arguments.Region).
+		Build()
+
+	if err != nil {
+		reporter.Errorf("Unable to build AWS client")
+		return err
+	}
+
+	regions, err := awsClient.DescribeRegions(&ec2.DescribeRegionsInput{})
+	if err != nil {
+		reporter.Errorf("Failed to describe regions")
+		return err
+	}
+
+	if zoneId != "" {
+		err := deleteHostedZoneId(awsClient, zoneId)
+		if err != nil {
+			reporter.Errorf("Unable to delete hosted zone: %s", err)
+		}
+		reporter.Infof("Hosted zone %s deleted", zoneId)
+	}
+
+	if zoneId == "" {
+		reporter.Infof("No zone id specified")
+		err := deleteAllHostedZones(reporter, logging, regions)
+		if err != nil {
+			reporter.Errorf("Unable to delete hosted zones: %s", err)
+		}
+	}
+
+	return
+}
+
+func init() {
+	flags := Cmd.Flags()
+	arguments.AddFlags(flags)
+	Cmd.Flags().StringVarP(&zoneId, "zone-id", "i", "", "Delete specific zone id")
+}
+
+func deleteHostedZoneId(client aws.Client, zoneId string) error {
+
+	input := &route53.DeleteHostedZoneInput{
+		Id: &zoneId,
+	}
+
+	_, err := client.DeleteHostedZonesByName(input)
+	if err != nil {
+		return fmt.Errorf("unable to delete hosted zone: %s", err)
+	}
+
+	return err
+}
+
+func deleteAllHostedZones(
+	reporter *rprtr.Object, logging *logrus.Logger, regions *ec2.DescribeRegionsOutput) error {
+	awsClient, err := aws.NewClient().
+		Logger(logging).
+		Profile(arguments.Profile).
+		RoleArn(arguments.RoleArn).
+		Build()
```

Comment:
we could pass in the same we already have in `run` instead of creating a new one